src/Common/src/Interop/Windows/BCrypt/Cng.cs

```diff
@@ -0,0 +1,394 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using global::System;
+using global::System.Text;
+using global::System.Diagnostics;
+using global::System.Diagnostics.Contracts;
+using global::System.Collections.Generic;
+using global::System.Security.Cryptography;
+using global::System.Runtime.InteropServices;
+
+namespace Internal.NativeCrypto
+{
+    internal static partial class BCryptNative
+    {
+        /// <summary>
+        ///     Well known algorithm names
+        /// </summary>
+        internal static class AlgorithmName
+        {
+            public const string ECDHP256 = "ECDH_P256";         // BCRYPT_ECDH_P256_ALGORITHM
+            public const string ECDHP384 = "ECDH_P384";         // BCRYPT_ECDH_P384_ALGORITHM
+            public const string ECDHP521 = "ECDH_P521";         // BCRYPT_ECDH_P521_ALGORITHM
+            public const string ECDsaP256 = "ECDSA_P256";       // BCRYPT_ECDSA_P256_ALGORITHM
+            public const string ECDsaP384 = "ECDSA_P384";       // BCRYPT_ECDSA_P384_ALGORITHM
+            public const string ECDsaP521 = "ECDSA_P521";       // BCRYPT_ECDSA_P521_ALGORITHM
+            public const string MD5 = "MD5";                    // BCRYPT_MD5_ALGORITHM
+            public const string Sha1 = "SHA1";                  // BCRYPT_SHA1_ALGORITHM
+            public const string Sha256 = "SHA256";              // BCRYPT_SHA256_ALGORITHM
+            public const string Sha384 = "SHA384";              // BCRYPT_SHA384_ALGORITHM
+            public const string Sha512 = "SHA512";              // BCRYPT_SHA512_ALGORITHM
+        }
+
+        /// <summary>
+        ///     Magic numbers identifying blob types
+        /// </summary>
+        internal enum KeyBlobMagicNumber {
+            ECDHPublicP256 = 0x314B4345,                        // BCRYPT_ECDH_PUBLIC_P256_MAGIC
+            ECDHPublicP384 = 0x334B4345,                        // BCRYPT_ECDH_PUBLIC_P384_MAGIC
+            ECDHPublicP521 = 0x354B4345,                        // BCRYPT_ECDH_PUBLIC_P521_MAGIC
+            ECDsaPublicP256 = 0x31534345,                       // BCRYPT_ECDSA_PUBLIC_P256_MAGIC
+            ECDsaPublicP384 = 0x33534345,                       // BCRYPT_ECDSA_PUBLIC_P384_MAGIC
+            ECDsaPublicP521 = 0x35534345                        // BCRYPT_ECDSA_PUBLIC_P521_MAGIC
+        }
+
+        internal static class KeyDerivationFunction
+        {
+            public const string Hash = "HASH";                  // BCRYPT_KDF_HASH
+            public const string Hmac = "HMAC";                  // BCRYPT_KDF_HMAC
+            public const string Tls = "TLS_PRF";                // BCRYPT_KDF_TLS_PRF
+        }
+    }
+
+    //
+    // Interop layer around Windows CNG api.
+    //
+    internal static partial class Cng
```

Comment:
In general there's probably some clean up to do in these interop files.  I'd suggest just opening an issue to deal with it separately.